hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableStakingInfoStoreImpl.java

```diff
@@ -46,4 +49,25 @@ public ReadableStakingInfoStoreImpl(@NonNull final ReadableStates states) {
     public StakingNodeInfo get(final long nodeId) {
         return stakingInfoState.get(nodeId);
     }
+
+    @NonNull
+    @Override
+    public Collection<StakingNodeInfo> getAll() {
+        // For entity types that have many instances this code would be a bad idea, but for node staking info there
+        // should only be a limited number of staking nodes in state. Iterating over all of them should not be expensive
+        final var keysIter = stakingInfoState.keys();
+        if (!keysIter.hasNext()) return Collections.emptySet();
+
+        final var nodeIds = new HashSet<Long>();
+        while (keysIter.hasNext()) {
+            nodeIds.add(keysIter.next());
+        }
+
+        final var infos = new HashSet<StakingNodeInfo>(nodeIds.size());
+        for (final long key : nodeIds) {
+            infos.add(stakingInfoState.get(key));
+        }
```

Comment:
I think we can do this logic in `EndOfStakingPeriodCalculator`

Reply:
Removed